### PR TITLE
Updates Permute doc with 3D example

### PR DESCRIPTION
**Pull request by mohantym**
Fixes  [#41123](https://github.com/tensorflow/tensorflow/issues/41123). 

1.  Removed compat.v2 
2. Added a new 3D example and updated Invalid argument error.

 As per MarkDaoust's[comment](https://github.com/tensorflow/tensorflow/issues/41123#issuecomment-991229661)

### CODE REVIEW

**Comment by gbaned** [comment]
@mohantym Can you please fix Lint failures ? Thank you!

File: keras/layers/reshaping/permute.py [hunk 1]
```diff
@@ -36,8 +36,8 @@ class Permute(Layer):
 
     ```python
     model = Sequential()
-    model.add(Permute((2, 1), input_shape=(10, 64)))
-    # now: model.output_shape == (None, 64, 10)
+    model.add(Permute((1, 3, 2), input_shape=(30, 6, 8)))
+    # now: model.output_shape == (None, 8, 6)
```
Comment (hertschuh) [inline keras/layers/reshaping/permute.py:40]: This should be `# now: model.output_shape == (None, 30, 8, 6)`

File: keras/layers/reshaping/permute.py [hunk 2]
```diff
@@ -62,9 +62,11 @@ def __init__(self, dims, **kwargs):
         self.dims = tuple(dims)
         if sorted(dims) != list(range(1, len(dims) + 1)):
             raise ValueError(
-                "Invalid permutation argument `dims` for Permute Layer. "
-                "The set of indices in `dims` must be consecutive and start "
-                f"from 1. Received dims={dims}"
+                "Invalid permutation `dims` for Permute Layer: %s. "
+                "`dims` must be a permutation of the input's axis ids, "
```
Comment (hertschuh) [inline keras/layers/reshaping/permute.py:66]: I would not refer to these as _axis ids_. These are simply the indices of the axes (as opposed to some identifiers).

How about:

> `dims` must be a permutation of the input's axis indices,

File: keras/layers/reshaping/permute.py [hunk 3]
```diff
@@ -62,9 +62,11 @@ def __init__(self, dims, **kwargs):
         self.dims = tuple(dims)
         if sorted(dims) != list(range(1, len(dims) + 1)):
             raise ValueError(
-                "Invalid permutation argument `dims` for Permute Layer. "
-                "The set of indices in `dims` must be consecutive and start "
-                f"from 1. Received dims={dims}"
+                "Invalid permutation `dims` for Permute Layer: %s. "
```
Comment (hertschuh) [inline keras/layers/reshaping/permute.py:65]: You're not providing the value for `%s`.
Also, it would be preferable to use an f-string like it was before.

How about:

```python
f"Permute layer received invalid permutation argument dims={dims}. "
```

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thank you for submitting this Pull Request!
Here are a couple of suggestions and fixes.

File: keras/layers/reshaping/permute.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 import copy
 
-import tensorflow.compat.v2 as tf
```
Comment (fchollet) [inline keras/layers/reshaping/permute.py:20]: Please revert this edit.

File: keras/layers/reshaping/permute.py (hunk 3 above)
Comment (fchollet) [reply keras/layers/reshaping/permute.py:65]: Also, please add a unit test with an `assertRaiseRegex` for this exception.

**Comment by alexpyattaev** [comment]
I think this addresses the issue https://github.com/tensorflow/tensorflow/issues/41123  only partially. The improved example is, of course, very nice. But the docs still do not mention that *all* dimensions must be listed. 
Also, I believe it will remain tricky for users to understand what the legal permutation list look like when just using the API. You have made the ValueError message even more cryptic than it was, as user would now absolutely have to read the source to figure out what the function expects. Maybe the longer form ValueError was not such a bad idea?
